### PR TITLE
Added support for scripts with no extension.

### PR DESCRIPTION
**Pull request by Azbagheri**
Also, Improved the tests error failures and organized the test data.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -19,46 +31,45 @@ process_input(){
...
 }
 
 scan_file(){
     local file_path=$1
     local file=$(basename -- "$file_path")
-    local extension="${file##*.}"
-
-    if [ "$extension" == "sh" ]; then
+    local first_line=$(head -n 1 "$file_path")
+    if [[ -x "$file_path" ]] && [[ "$first_line" == "#!"* ]]; then
```
Comment (asadmansr) [inline entrypoint.sh]: `-x "$file_path"`

Why are we checking if a file is executable? Would the `#!` check be enough

Context: I had a few shell scripts that weren't executable, so I was just wondering. Also, I made a new file `temp` without an extension, made it executable and did not add the shebang. Shellcheck reported an error. So I think the shebang should be enough

File: entrypoint.sh [hunk 2]
```diff
@@ -19,46 +31,45 @@ process_input(){
...
 }
 
 scan_all(){
-    echo "Scanning all the scripts with format name.sh at $1 🔎"
+    echo "Scanning all the shell scripts at $1 🔎"
     while IFS= read -r -d $'\0' script 
     do
         scan_file "$script"
-    done < <(find "$1" -name '*.sh' -print0)
+    done < <(find "$1" -type f -perm ${perm_arg} ! -path "$1/.git/*" -print0 )
```
Comment (asadmansr) [inline entrypoint.sh]: For scan all, we are finding all the files in a given folder ($1). Why are we specifying the path?

File: README.md [hunk 3]
```diff
@@ -47,6 +46,13 @@ Run static analysis for all the shell scripts in a folder.
           path: "src"
 ```
 
+You can even use **wildcard** in the path
```
[the remark this answers is not in the document]
Comment (Azbagheri) [reply README.md]: sure, I like that!

File: entrypoint.sh [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
 my_dir=$(pwd)
-VERSION="0.1.0"
 status_code="0"
 
+if [[ "$OSTYPE" == "linux"* ]]; then
+    perm_arg=/111
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    perm_arg=+111
+else
+    echo "Operating system $OSTYPE is not supported."
```
[the remark this answers is not in the document]
Comment (Azbagheri) [reply entrypoint.sh]: God point. I'll change the echo to say what OS are supported.

File: entrypoint.sh [hunk 5]
```diff
@@ -1,11 +1,23 @@
 #! /bin/bash
+# shellcheck disable=SC2155
```
[the remark this answers is not in the document]
Comment (Azbagheri) [reply entrypoint.sh]: This is related to shellcheck comments. If I remove it, the script won't pass shellcheck. Very safe to disable that warning though.

File: entrypoint.sh [hunk 6]
```diff
@@ -19,46 +31,45 @@ process_input(){
...
         if [ $exit_code -eq 0 ] ; then
-            echo "Successfully scanned ${file_path} 🙌"
+            printf "%b" "Successfully scanned ${file_path} 🙌\n"
         else
             status_code=$exit_code
             printf "\e[31m \n ERROR: ShellCheck detected issues in %s.\e[0m \n" "${file_path} 🐛"
         fi
     else
-        printf "\e[33m ⚠️  Warning: invalid file extension. Make sure the input file '%s' is a valid shell script.\e[0m\n" "$file"
+        printf "\e[33m ⚠️  Warning: '%s' is not a valid shell script. Make sure file is executable & shebang is on the first line.\e[0m\n" "$file_path"
```
[the remark this answers is not in the document]
Comment (Azbagheri) [reply entrypoint.sh]: :)

File: entrypoint.sh (hunk 2 above)
Comment (Azbagheri) [reply entrypoint.sh]: Now that we lint scripts with no extension, the default case will lint all the scripts in the .git directory as well. And we don't want that, that's why I'm excluding the `.git` directory.

**Pull request merged**